### PR TITLE
[stable/concourse] Specify secrets mount path

### PR DESCRIPTION
**Pull request by goddenrich**
#### What this PR does / why we need it:
This PR allows the user to specify where the secrets that manage the system are mounted on the web and worker agent. For our security tools we need to specify where the secrets are found or expected.

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md

### CODE REVIEW

**Comment by william-tran** [comment]
Could you provide some more detail on this security tool you're using? I'm afraid without such details this would be a feature that would only be useful to a use case not shared by many. Also, `secrets.*` should be actual secret values, so I'd find another key for that.

**Comment by goddenrich** [comment]
> Could you provide some more detail on this security tool you're using? I'm afraid without such details this would be a feature that would only be useful to a use case not shared by many. Also, `secrets.*` should be actual secret values, so I'd find another key for that.

We are using hashicorp vault to manage all secrets and we have a tool that manages them so we need a way to control where the secrets are found.

I see what you mean about the secrets key so will change them

**Comment by goddenrich** [comment]
@william-tran I have used a different key. I think this makes more sense where it does than before

**Comment by cirocosta** [comment]
Hey @goddenrich, thanks for submitting this PR!

**Do you know if there are any other charts that follow this convention?**

I've seen charts like Prometheus and Grafana going through the `extraSecretMounts` approach which seems pretty flexible in terms of both mount locations and supplying your own secrets:

https://github.com/helm/charts/blob/353ba5ef6467fd64035b7d5446df426f86d60153/stable/grafana/templates/deployment.yaml#L59-L63

https://github.com/helm/charts/blob/61c28625ac5034f908b4b49f47f3723cf2bc3348/stable/prometheus/templates/server-deployment.yaml#L136-L141

I definitely like the idea of being able to specify secrets that have been created outside the chart itself, but I'm a bit worried about the conventions (having just started using Helm myself).

Wdyt?

thx!

**Comment by goddenrich** [comment]
@william-tran any further comments?

**Comment by william-tran** [comment]
Thanks for your patience @goddenrich, I'm good with these changes. @cirocosta how about you? Because we need to correlate the mount config with env vars I can see the benefit in supporting each of those things directly in the chart vs. a more generic approach so we are DRY. 

You're up next in the PR list: #10617 please rebase and bump the chart to 3.2.0

**Comment by william-tran** [comment]
You'll need to rebase, otherwise looks good

File: stable/concourse/values.yaml [hunk 1]
```diff
@@ -665,6 +665,13 @@ web:
   #   - name: CONCOURSE_TSA_LOG_LEVEL
   #     value: "debug"
 
+  ## For managing where secrets should be mounted for the web agents
```
Comment (paulczar) [inline stable/concourse/values.yaml:697]: why are you commenting these out and then using defaults in the templates ?

**Comment by cirocosta** [comment]
It sounds pretty good!

With https://github.com/helm/charts/pull/9665 in, I suppose this PR should be updated to include those secrets too.

thx!

File: stable/concourse/values.yaml (hunk 1 above)
Comment (william-tran) [reply stable/concourse/values.yaml:697]: Would a better approach be to remove the call to default in the templates and just set the default values here?

Comment (goddenrich) [reply stable/concourse/values.yaml:697]: sure sounds better

**Review by william-tran — APPROVED** [review]

**Comment by goddenrich** [comment]
Thanks guys! Does the hold tag need to be removed?

**Pull request merged**